src/printer.js

```diff
@@ -1790,6 +1790,10 @@ function printFunctionParams(path, print, options) {
     printed.push(concat([ "...", path.call(print, "rest") ]));
   }
 
+  if (printed.length === 0) {
+    return "()";
+  }
+
   return concat([
     "(",
     indent(
```

Comment:
Yep, this is the easiest thing to do! The representation below can't completely collapse in this case because it wants to break across lines. But because we can statically analyze this and easily know we never want to break up the parens your solution is correct.